changelog.d/11602.bugfix

```diff
@@ -0,0 +1 @@
+Fix a long-standing bug that some unknown endpoints would return HTML 404 errors instead of JSON 400 errors.
```

Comment:
I'd probably change to this:
```suggestion
Fix a long-standing bug that some unknown endpoints would return HTTP 404 errors instead of JSON `M_UNRECOGNIZED` errors.
```

Reply:
Errr, yes. Too many abbreviations in my head! Thanks!

Reply:
No, this was right the way it was. The point of this PR is to return JSON errors instead of HTML ones.